hail/python/hailtop/hailctl/config/initialize.py

```diff
@@ -0,0 +1,240 @@
+            verbose=verbose
+        )
+    else:
+        remote_tmpdir, warnings = await setup_existing_remote_tmpdir(region, hail_identity, verbose)
+
+    return (remote_tmpdir, region, warnings)
+
+
+async def async_basic_initialize(incremental: bool = False, verbose: bool = False):
+    from hailtop.auth.auth import async_get_userinfo  # pylint: disable=import-outside-toplevel
+    from hailtop.batch_client.aioclient import BatchClient  # pylint: disable=import-outside-toplevel
+    from hailtop.config import DeployConfig  # pylint: disable=import-outside-toplevel
+    from hailtop.config import get_user_config_path  # pylint: disable=import-outside-toplevel
+
+    from ..auth.login import async_login  # pylint: disable=import-outside-toplevel
+    from .cli import set as set_config, list as list_config  # pylint: disable=import-outside-toplevel
+    from .utils import already_logged_into_service, get_default_region  # pylint: disable=import-outside-toplevel
+
+    domain = Prompt.ask('What domain is the Hail service running in?', default='hail.is')
+    namespace = 'default'
+
+    deploy_config = DeployConfig('external', namespace, domain)
+    deploy_config.dump_to_file()
+
+    already_logged_in = await already_logged_into_service()
+    if not already_logged_in:
+        await async_login(namespace)
+
+    user_info = await async_get_userinfo()
+    username = user_info['username']
+    hail_identity = user_info['hail_identity']
+    trial_bp_name = user_info['trial_bp_name']
+
+    batch_client = await BatchClient.create(trial_bp_name, deploy_config=deploy_config)
+
+    async with batch_client:
+        cloud = await batch_client.cloud()
+        supported_regions = await batch_client.supported_regions()
+        default_region = get_default_region(supported_regions, 'gcp')
+
+        if cloud == 'gcp':
+            remote_tmpdir, region, warnings = await initialize_gcp(username, hail_identity, supported_regions, default_region, verbose)
+        else:
+            region = Prompt.ask('Which region should resources be created in?', default=default_region,
+                                choices=supported_regions)
+
+            remote_tmpdir = Prompt.ask(f'Enter a path to an existing remote temporary directory (ex: gs://my-bucket/batch/tmp)')
+            typer.secho(f'WARNING: You will need to grant read/write access to {remote_tmpdir} for account {hail_identity}', fg=typer.colors.MAGENTA)
+            warnings = True
+
+    config_file = get_user_config_path()
+
+    if not incremental:
+        if os.path.isfile(config_file):
+            os.remove(config_file)
+
+    set_config('domain', domain)
+
+    if trial_bp_name:
+        set_config('batch/billing_project', trial_bp_name)
+
+    if remote_tmpdir:
+        set_config('batch/remote_tmpdir', remote_tmpdir)
+
+    set_config('batch/regions', region)
+    set_config('batch/backend', 'service')
+    set_config('query/backend', 'batch')
```

Comment:
I think this is a bit bold of a move at present. I feel like we should at least prompt the user to ask which backend they want to use for query.